### ISSUE TITLE
[Benefits Documents] Update param passed for tracked item id

### ISSUE THREAD

**Issue by jerekshoe** [department-of-veterans-affairs/va.gov-team#72886]
## Background
There was a change to the Benefits Documents Service in regards to the names of the param that it uses for the id of the tracked item that is associated with the document upload (if there is one). We are passing a value of `trackedItemId`, when we should be passing `trackedItemIds`. We also need to wrap the value being passed in an array

## Problem
Because we are passing the wrong param name and value type, documents uploaded to the LH Benefits Documents service to satisfy tracked item requirements don't actually get associated to those tracked items properly (they do get uploaded successfully though). This means that when a Veteran has a tracked item and they upload a document to satisfy it, the tracked item doesn't update like it should

## Tasks
- [x] Update the param passed to `tracked_item_ids`
- [x] Wrap the value of the param in an array and apply `JSON.stringify`
- [x] Parse the JSON in vets-api to an array of Numbers

## Acceptance Criteria
- [x] The `trackedItemId` param key and value have been updated to conform to the requirements of the `BenefitsDocuments::Service`
- [x] When files are uploaded to satisfy tracked items, those files are properly associated with that tracked item (i.e. their trackedItemId matches the tracked items id)
- [x] When files are uploaded to satisfy a tracked item, the tracked items status changes from NEEDED_FROM_YOU to SUBMITTED_AWAITING_REVIEW

**Comment by jerekshoe** [comment]
**Update (1/5):** PR Is out for this. Testing with benefits documents stuff locally, will test in staging to make sure it works as expected. None of the code for this is used in prod yet so it should be safe to wait till it's deployed to test

**Comment by jerekshoe** [comment]
**Update (1/8):** Initial PR didn't quite work as expected. The parameter sent by vets-website to vets-api gets converted into a regular string, even if the initial value was an array. So this:
```js
tracked_item_ids: [123, 456]
```
is seen as this in vets-api:
```ruby
params[:tracked_item_ids]: "123,456"
```
This leaves us with 2 options:
1) Convert `params[:tracked_item_ids]` in vets-api manually i.e.:
```ruby
params[:tracked_item_ids] = Array.wrap(params[:tracked_item_ids].split(','))
```
#### or 
2) we can stringify the array in vets-website before sending it to vets-api:
```js
tracked_item_ids: JSON.stringify([trackedItemId)])
```
AND parse the JSON in vets-api:
```ruby
params[:tracked_item_ids] = JSON.parse(params[:tracked_item_ids])
```

I'm leaning towards option 2 as it seems more straightforward than option 1

**Comment by jerekshoe** [comment]
## Update (1/9)
Backend PR is merged, waiting on reviews for the FE PR. Once both are merged into staging, I plan on testing again to make sure that this is working as expected

**Comment by HeatherWidmont** [comment]
@pmclaren19 will review today before she leaves for PTO

**Comment by jerekshoe** [comment]
### Update (1/9)
All PRs created so far have been merged, unfortunately I'm not able to determine if the BE has been deployed yet (BE deploy dashboard is down currently). Was still getting 400s earlier, but I'm not sure if that's because of an error with the latest code, or if it's just because the change hasn't deployed yet

**Comment by jerekshoe** [comment]
### Update (1/9)
The Backend PR hasn't deployed to staging yet, keeping track of the deployment status [here](https://deploy-status-dashboard.fly.dev/apps/vets-api/deploy_status). Will test to see if this is working as expected or not when the deploy has succeeded

**Comment by jerekshoe** [comment]
### Update (1/10)
The Backend PR deployed to staging and I just finished testing it. From what I can tell it appears to be working as I was able to successfully upload a document to satisfy an evidence request and 1) The upload was successful and 2) The tracked item now has a status of "SUBMITTING_AWAITING_REVIEW"

**Uploaded document has a non-null trackedItemId**
![Screenshot 2024-01-10 at 2 34 51 PM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/abbade1e-b2d1-40f0-9d4f-b159b29176ca)
---
**Tracked item showing status of "SUBMITTED AWAITING REVIEW"**
![Screenshot 2024-01-10 at 2 35 05 PM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/13838621/60b80ac3-c869-4553-91c0-96f002ee9e43)
